yt_dlp/extractor/dw.py

```diff
@@ -1,111 +1,164 @@
         webpage = self._download_webpage(url, media_id)
-        hidden_inputs = self._hidden_inputs(webpage)
-        title = hidden_inputs['media_title']
-        media_id = hidden_inputs.get('media_id') or media_id
 
-        direct_url = url_or_none(hidden_inputs.get('file_name'))
-        if direct_url:
-            formats = [{'url': hidden_inputs['file_name']}]
-        else:
-            formats = self._extract_smil_formats(
-                f'http://www.dw.com/smil/v-{media_id}', media_id,
-                transform_source=lambda s: s.replace(
-                    'rtmp://tv-od.dw.de/flash/',
-                    'http://tv-download.dw.de/dwtv_video/flv/'))
+        app_state = self._search_json(
+            r'window\.__APP_STATE__\s*=\s*', webpage, 'app state', media_id, default={})
+        if not app_state:
+            title = self._html_search_meta('twitter:title', webpage)
+            pattern = re.compile(r'<source[^>]+src\s*=\s*(["\'])(?P<url>.+?)\1')
+            entries = [{
+                'id': url_basename(m.group('url')).removesuffix('.mp3'),
+                'ext': 'mp3',
+                'title': title,
+                'url': m.group('url'),
+                'vcodec': 'none',
+            } for m in pattern.finditer(webpage)]
 
-        upload_date = hidden_inputs.get('display_date')
-        if not upload_date:
-            upload_date = self._html_search_regex(
-                r'<span[^>]+class="date">([0-9.]+)\s*\|', webpage,
-                'upload date', default=None)
-            upload_date = unified_strdate(upload_date)
+            return self.playlist_result(entries, media_id, title)
+
+        path = self._PATH_MAP[media_type]
+        graph_api = traverse_obj(app_state, (
+            f'/graph-api/{lang}/content/{path}/{media_id}', 'data', 'content', {dict}))
+
+        if media_type in self._ENTRIES_PATH_MAP:
+            return self.playlist_result(
+                self._entries(url, graph_api, media_type), media_id, graph_api.get('title'))
+        elif media_type == 'audio':
+            formats = [{
+                'ext': 'mp3',
+                'url': traverse_obj(graph_api, 'mp3Src', {url_or_none}),
+                'vcodec': 'none',
+            }]
+        else:
+            m3u8_url = traverse_obj(graph_api, 'hlsVideoSrc', {url_or_none})
+            formats = self._extract_m3u8_formats(m3u8_url, media_id, 'mp4', m3u8_id='hls')
 
         return {
             'id': media_id,
-            'title': title,
-            'description': self._og_search_description(webpage),
-            'thumbnail': hidden_inputs.get('preview_image'),
-            'duration': int_or_none(hidden_inputs.get('file_duration')),
-            'upload_date': upload_date,
             'formats': formats,
+            **traverse_obj(graph_api, {
+                'title': ('title', {str}),
+                'categories': ('thematicFocusCategory', 'name', {str}, all, filter),
```

Comment:
`filter` after `all` will filter out an empty list; `filter` before `all` will filter out empty strings from the list
```suggestion
                'categories': ('thematicFocusCategory', 'name', {str}, filter, all, filter),
```